### PR TITLE
golang: bump automation with updatecli and GH actions

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Automate the golang bump within the project itself using [updatecli](https://www.updatecli.io/) and github actions

## Why is it important?

Remove the dependency with Jenkins and the definition of what to do in [apm-pipeline-library](https://github.com/elastic/apm-pipeline-library/blob/a3850732518748f2e71815c093d58cb071d0d8a3/.ci/.bump-go-release-version.yml#L33-L40)

Use updatecli that provides the tooling for bumping versions

## Issues

Similar to https://github.com/elastic/beats/pull/34458

## Test

```
GIT_USER=*** \
GIT_EMAIL=*** \
updatecli diff --config .ci/bump-golang.yml
```

does nothing in terms of changing if no a new version.

if a new version is available then it bumps all those files as expected

<details><summary>Expand to view</summary>
<p>

```
+++++++++++
+ PREPARE +
+++++++++++

Loading Pipeline ".ci/bump-golang.yml"

SCM repository retrieved: 1



++++++++++++++++++
+ AUTO DISCOVERY +
++++++++++++++++++



++++++++++++
+ PIPELINE +
++++++++++++



#########################################
# BUMP GOLANG-VERSION TO LATEST VERSION #
#########################################


SOURCES
=======

minor
-----
The shell 🐚 command "cat .go-version" ran successfully with the following output:
----
1.18.6
----

latestGoVersion
---------------
WARNING: ⚠ No GitHub Release found, we fallback to published git tags
Searching for version matching pattern "go1\\.18\\.(\\d*)$"
✔ Github Release version "go1.18.10" found matching pattern "go1\\.18\\.(\\d*)$"


CHANGELOG:
----------
no GitHub Release found for go1.18.10 on "https://github.com/golang/go"

gomod
-----
The shell 🐚 command "echo 1.18.10" ran successfully with the following output:
----
1.18.10
----


CONDITIONS:
===========

goDefaultVersion-check
----------------------
The shell 🐚 command "grep -v -q 1.18.10 .go-version # 1.18.10" ran successfully with the following output:
----
----

dockerTag
---------
✔ The Docker image index.docker.io/library/golang:1.18.10 (amd64) exists and is available.


TARGETS
========

update-version.asciidoc
-----------------------

**Dry Run enabled**

⚠ updated the [dry run] content of the file "/var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/updatecli/github/elastic/cloudbeat/docs/version.asciidoc"

--- /var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/updatecli/github/elastic/cloudbeat/docs/version.asciidoc
+++ /var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/updatecli/github/elastic/cloudbeat/docs/version.asciidoc
@@ -1,6 +1,6 @@
 // doc-branch can be: master, 8.1, 8.2, etc.
 :doc-branch: master
-:go-version: 1.18.6
+:go-version: 1.18.10
 :python: 3.7
 :docker: 1.12
 :docker-compose: 1.11


update-go-version
-----------------

**Dry Run enabled**

⚠ updated the [dry run] content of the file "/var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/updatecli/github/elastic/cloudbeat/.go-version"

--- /var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/updatecli/github/elastic/cloudbeat/.go-version
+++ /var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/updatecli/github/elastic/cloudbeat/.go-version
@@ -1 +1 @@
-1.18.6
+1.18.10


update-gomod
------------

**Dry Run enabled**

✔ Content from file "/var/folders/hn/j6gds1qx1f5bysfn0qp0t03c0000gn/T/updatecli/github/elastic/cloudbeat/go.mod" already up to date
✔ All contents from 'file' and 'files' combined already up to date


ACTIONS
========

WARNING: **Fallback** Please add a title to you configuration using the field 'title: <your pipeline>'
[Dry Run] An action of kind "github/pullrequest" is expected.

=============================

REPORTS:


⚠ Bump golang-version to latest version:
	Source:
		✔ [gomod] Get version in go.mod format (kind: shell)
		✔ [latestGoVersion] Get Latest Go Release (kind: githubrelease)
		✔ [minor] Get minor version in .go-version (kind: shell)
	Condition:
		✔ [dockerTag] Is docker image golang:1.18.10 published (kind: dockerimage)
		✔ [goDefaultVersion-check] Check if defined golang version differs (kind: shell)
	Target:
		⚠ [update-go-version] Update .go-version (kind: file)
		✔ [update-gomod] Update go.mod (kind: file)
		⚠ [update-version.asciidoc] Update version.asciidoc (kind: file)


Run Summary
===========
Pipeline(s) run:
  * Changed:	1
  * Failed:	0
  * Skipped:	0
  * Succeeded:	0
  * Total:	1

```

</p>
</details>

https://github.com/elastic/beats/pull/34486 was created to test these changes

### CODE REVIEW

**Review by reakaleek — APPROVED** [review]

**Review by oren-zohar — APPROVED** [review]

**Review by amannocci — APPROVED** [review]

File: .github/workflows/bump-golang.yml
Comment (v1v) [inline .github/workflows/bump-golang.yml:5]:
```suggestion
  workflow_dispatch:
  schedule:
```

**Review by amannocci — APPROVED** [review]

**Comment by v1v** [comment]
@oren-zohar , I'm afraid I cannot merge this PR, if you can help me with :)

**Comment by oren-zohar** [comment]
@v1v seems like this isn't working https://github.com/elastic/cloudbeat/actions/runs/4182824897/jobs/7246407903

**Comment by v1v** [comment]
```
WARNING: **Fallback** Please add a title to you configuration using the field 'title: <your pipeline>'
Pipeline "Bump golang-version to latest version" failed
Skipping due to:
	action stage:	"GitHub Actions is not permitted to create or approve pull requests"
```

I'll update it to use a service account instead

**Comment by v1v** [comment]
@oren-zohar -> https://github.com/elastic/cloudbeat/pull/749 should help with, but there is a hard requirement to add those GH secrets

**Pull request merged**
